git/init.go

```diff
@@ -111,5 +120,42 @@ func Init(c *Client, opts InitOptions, dir string) (*Client, error) {
 	}
 	c.WorkDir = WorkDir(wd)
 	c.GitDir = GitDir(wd + "/.git")
+
+	if opts.Template != "" {
+		err = filepath.Walk(opts.Template, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+                        path, err = filepath.Rel(opts.Template, path)
+                        if err != nil {
+                                return err
+                        }
+			if info.IsDir() {
+				os.MkdirAll(filepath.Join(c.GitDir.String(), path), 0770)
```

Comment:
My review had a comment here that somehow got eaten.. I think this should be 0777. The OS umask takes care of unsetting the bits that aren't supposed to be set (I made a test program that ran `os.MkdirAll("foo", 0777)` and foo came out 0755 to be sure that MkdirAll uses the umask, since the docs don't mention it..)